app/src/hooks/useFilterDelays.js

```diff
@@ -0,0 +1,85 @@
+import { useState, useEffect, useCallback } from 'react'
+import { useCurrentApp } from '@aragon/api-react'
+
+import STATUS from '../delay-status-types'
+
+const NULL_FILTER_STATE = -1
+const STATUS_FILTER_ONGOING = 1
+const STATUS_FILTER_PAUSED = 2
+const STATUS_FILTER_PENDING = 3
+const APP_FILTER_THIS = 1
+
+function testStatusFilter(filter, scriptStatus) {
+  return (
+    filter === NULL_FILTER_STATE ||
+    (filter === STATUS_FILTER_ONGOING && scriptStatus === STATUS.ONGOING) ||
+    (filter === STATUS_FILTER_PAUSED && scriptStatus === STATUS.PAUSED) ||
+    (filter === STATUS_FILTER_PENDING && scriptStatus === STATUS.PENDING_EXECUTION)
+  )
+}
+
+function testAppFilter(filter, script, { apps, thisAddress }) {
```

Comment:
should we call that function as testAppFilter?

Reply:
Will change to `checkAppFilter` (same for `testStatusFilter`)